src/lib/core/components/GlobalFiltersDialog.tsx

```diff
@@ -0,0 +1,118 @@
+import { StudyEntity } from '..';
+import { Filter } from '../types/filter';
+import FilterChipList from './FilterChipList';
+import Draggable from 'react-draggable';
+import { ResizableBox } from 'react-resizable';
+import { Alert } from '@material-ui/lab';
+import _ from 'lodash';
+import 'react-resizable/css/styles.css';
+import './GlobalFiltersDialog.scss';
+
+interface Props {
+  // A function to set whether the dialog is open
+  setOpen: (open: boolean) => void;
+  // The list of entities over which filters are being applied
+  entities: StudyEntity[];
+  // The list of filters to display
+  filters: Filter[];
+  // A function to set the filters
+  setFilters: (filters: Filter[]) => void;
+  // A function to remove a given filter
+  removeFilter: (filter: Filter) => void;
+}
+
+/**
+ * A dialog that displays filters sorted into bins by entity.
+ */
+export default function GlobalFiltersDialog(props: Props) {
+  let content: JSX.Element;
+
+  if (props.filters.length > 0) {
+    // Construct a filter chip list for each entity that has filters applied
+    let filters = props.filters; // The remaining unbinned filters
+    let matchingFilters: Filter[] = [];
+    let filterChipLists: JSX.Element[] = [];
+
+    for (const entity of props.entities) {
```

Comment:
Just as an FYI, in cases like these, it's more idiomatic to use [Array.prototype.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) instead of a `for` loop.

("cases like these" = "cases where you're iterating through each `element` of an `array` and updating/mutating some sort of `value` based on each `element`" In this case, you're iterating through each `entity` and updating `filters`, `matchingFilters`, and `filterChipLists` based on each of those `entity`s.)

I don't think it's necessary to hold up the review/merge over this, so after the merge, I'll leave a comment here showing how this `for` loop could have been refactored to use `reduce`. That'll make for a good reference for how you could use `reduce` in future work!

Reply:
https://www.youtube.com/watch?v=qaGjS7-qWzg ;-)

Reply:
Thanks for the suggestion, Jamie! I'm familiar with `reduce`. Bob's counterpoint (implied, in video form) is worth considering. I think it's an interesting question as to whether `reduce` in this instance would improve the code. I'm open!